aten/src/ATen/CMakeLists.txt

```diff
@@ -156,11 +156,10 @@ file(GLOB native_transformers_hip_cpp "native/transformers/hip/*.cpp")
 file(GLOB native_quantized_cudnn_hip_cpp "native/quantized/cudnn/hip/*.cpp")
 file(GLOB native_utils_cpp "native/utils/*.cpp")
 
-# flash_attention sources
-file(GLOB flash_attention_cuda_cu "native/transformers/cuda/flash_attn/*.cu")
-file(GLOB flash_attention_cuda_cpp "native/transformers/cuda/flash_attn/*.cpp")
-
-if(USE_FLASH_ATTENTION)
+if(USE_CUDA AND NOT USE_ROCM)
```

Comment:
Why are we no longer using the USE_FLASH_ATTENTION flag to gate whether or not we build the sources?

Reply:
To enable the code at runtime it needs to have been compiled. It'll be compiled for NVIDIA CUDA only.

Reply:
But I thought we can't build flash unless we are on an A100? So CI is going to complain

Reply:
Got it, I thought for some reason we weren't able to compile this code when not on both modern CUDA and newer cuda arch. I wonder if nvcc flags specifying arch will not work for CI/CD

Reply:
Yeah we might need to update those flags so that CI builds for 8.0 as well. Let's wait for build failures again before we do that.

Reply:
Perhaps we can still use this flag, but just set it to TRUE by default in a top-level CMakeLists.txt? (When option is defined)